### DISCUSSION TITLE
GTM, Pixels Tracking Plugins with Partytown

### DISCUSSION THREAD

**Discussion by shalawfatah** [gatsbyjs/gatsby#36598] [Ideas / Feature Requests]
The official tracking plugins like Google Analytics, Google Tracking Manager, or Facebook Pixels, are all outdated. There should be a new implementation where `Gatsby Script` is used with `Partytown`, or the plugin should give options like `idle, out-of-the-main-thread, etc`. I feel like this is very important for businesses using Gatsby headless.

**Comment by jawakarD** [comment]
Was going to create a discussion for this, but found this.

My proposal was to add a `strategy` option to the plugins `gatsby-plugin-google-tagmanager` and `gatsby-plugin-google-gtag`. And use Script API for implementation. 

I'll also try to open a PR to support this.
